app/src/main/java/com/android/bookswap/model/userViewModel.kt

```diff
@@ -157,4 +159,14 @@ open class UserViewModel(
       }
     }
   }
+  /**
+   * Disconnect the current user by resetting the user data and UUID. This method sets the user data
+   * to a new DataUser instance with a new UUID, marks the user as not loaded, and updates the
+   * stored state to false.
+   */
+  fun disconnectUser() {
+    this.dataUser = DataUser(UUID.randomUUID())
+    this.uuid = UUID.randomUUID()
+    Firebase.auth.signOut()
```

Comment:
The comment says that it marks the user as not loaded but the function doesn't change the isLoaded or the _isStored. Also it should update the user if the _isStored is false before disconnecting the user, to avoid lose of data.

Reply:
Also the dataUser should take in argument the new "this.uuid" that is created with the randomUUID(), right now they use different UUID which is strange.